src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseSimulationService.java

```diff
@@ -0,0 +1,157 @@
+            String solutionRepoName) {
+        final var projectKey = programmingExercise.getProjectKey();
+        final var templateParticipation = programmingExercise.getTemplateParticipation();
+        final var solutionParticipation = programmingExercise.getSolutionParticipation();
+        final var templatePlanName = TEMPLATE.getName();
+        final var solutionPlanName = SOLUTION.getName();
+        final var exerciseRepoUrl = "http://localhost:7990/scm/" + projectKey + "/" + exerciseRepoName + ".git";
+        final var testsRepoUrl = "http://localhost:7990/scm/" + projectKey + "/" + testRepoName + ".git";
+        final var solutionRepoUrl = "http://localhost:7990/scm/" + projectKey + "/" + solutionRepoName + ".git";
+        templateParticipation.setBuildPlanId(projectKey + "-" + templatePlanName);
+        templateParticipation.setRepositoryUrl(exerciseRepoUrl);
+        solutionParticipation.setBuildPlanId(projectKey + "-" + solutionPlanName);
+        solutionParticipation.setRepositoryUrl(solutionRepoUrl);
+        programmingExercise.setTestRepositoryUrl(testsRepoUrl);
+    }
+
+    /**
+     * This method creates the template and solution submissions and results for the new exercise
+     * These submissions and results are SIMULATIONS for the testing of programming exercises without local setup
+     * @param programmingExercise the new exercise
+     */
+    public void setupInitialSubmissionsAndResults(ProgrammingExercise programmingExercise) {
+        Optional<TemplateProgrammingExerciseParticipation> templateProgrammingExerciseParticipation = this.templateProgrammingExerciseParticipationRepository
+                .findByProgrammingExerciseId(programmingExercise.getId());
+        Optional<SolutionProgrammingExerciseParticipation> solutionProgrammingExerciseParticipation = this.solutionProgrammingExerciseParticipationRepository
+                .findByProgrammingExerciseId(programmingExercise.getId());
+        String commitHashBase = "abcdef01";
+        ProgrammingSubmission programmingSubmissionBase = new ProgrammingSubmission();
+        programmingSubmissionBase.setParticipation(templateProgrammingExerciseParticipation.get());
+        programmingSubmissionBase.setSubmitted(true);
+        programmingSubmissionBase.setType(SubmissionType.OTHER);
+        programmingSubmissionBase.setCommitHash(commitHashBase);
+        programmingSubmissionBase.setSubmissionDate(templateProgrammingExerciseParticipation.get().getInitializationDate());
+        programmingSubmissionRepository.save(programmingSubmissionBase);
+        Result resultBase = new Result();
+        resultBase.setParticipation(templateProgrammingExerciseParticipation.get());
+        resultBase.setSubmission(programmingSubmissionBase);
+        resultBase.setRated(true);
+        resultBase.resultString("13 of 13 failed");
```

Comment:
I will address this in a different PR for the other occurences, but we should change this to `0 of 13 passed` to be consistent with this: https://github.com/ls1intum/Artemis/blob/044f7aacb64823a1d9838ada2dd31e2ca7951c16/src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseTestCaseService.java#L251

This is not a problem introduced by your PR, but we should change it.

Reply:
I adpated the naming scheme